Lib/filecmp.py

```diff
@@ -185,14 +187,15 @@ def phase3(self): # Find out differences between common files
         self.same_files, self.diff_files, self.funny_files = xx
 
     def phase4(self): # Find out differences between common subdirectories
-        # A new dircmp object is created for each common subdirectory,
+        # A new dircmp (or MyDirCmp if dircmp was subclassed) object is created
+        # for each common subdirectory,
         # these are stored in a dictionary indexed by filename.
         # The hide and ignore properties are inherited from the parent
         self.subdirs = {}
         for x in self.common_dirs:
             a_x = os.path.join(self.left, x)
             b_x = os.path.join(self.right, x)
-            self.subdirs[x]  = dircmp(a_x, b_x, self.ignore, self.hide)
+            self.subdirs[x]  = self.__class__(a_x, b_x, self.ignore, self.hide)
```

Comment:
Why not use `type(self)` here?

Reply:
Not really any good reason, I was copying @cjerdonek's patch and that's what he used. They evaluate to the same thing, right? I believe in Python 2 we would have needed to use `.__class__`, but now there is not that restriction. You're just thinking of style?

Reply:
The same thing I guess, both variants are equal.